chainer/functions/normalization/batch_normalization.py

```diff
@@ -27,130 +66,294 @@ def check_type_forward(self, in_types):
 
     Args:
         x (Variable): The input variable.
         gamma (Variable): The scaling parameter of normalized data.
         beta (Variable): The shifting parameter of scaled normalized data.
         eps (float): Epsilon value for numerical stability.
+        running_mean (array): The running average of the mean. This is a
+        running average of the mean over several
+            mini-batches using the decay parameter. If None, the running
+            average is not computed. If this is None,
+            then runnng_var must also be None.
+        running_var (array): The running average of the variance. This is a
+        running average of the variance
+            over several mini-batches using the decay parameter. If None, the
+            running average is not computed. If
+            this is None, then running_mean must also be None.
+        decay (float): Decay rate of moving average. It is used during
+            training.
+
 
     See: `Batch Normalization: Accelerating Deep Network Training by Reducing\
           Internal Covariate Shift <http://arxiv.org/abs/1502.03167>`_
 
     .. seealso:: :class:`links.BatchNormalization`
 
     """
-    return BatchNormalizationFunction(eps)(x, gamma, beta)
+    return BatchNormalizationFunction(eps, running_mean, running_var, True,
+                                      decay, use_cudnn)(x, gamma, beta)
 
 
-def fixed_batch_normalization(x, gamma, beta, mean, var, eps=1e-5):
+def fixed_batch_normalization(x, gamma, beta, mean, var, eps=2e-5,
+                              use_cudnn=True):
     """Batch normalization function with fixed statistics.
 
     This is a variant of batch normalization, where the mean and variance
-    statistics are given by the caller as variables. This is used on testing
-    mode of the batch normalization layer, where batch statistics cannot be
-    used for prediction consistency.
+    statistics are given by the caller as fixed parameters. This is
+    used on testing mode of the batch normalization layer, where batch
+    statistics cannot be used for prediction consistency.
 
     Args:
         x (Variable): The input variable.
         gamma (Variable): The scaling parameter of normalized data.
         beta (Variable): The shifting parameter of scaled normalized data.
-        mean (Variable): The shifting parameter of input.
-        var (Variable): The square of scaling parameter of input.
+        mean (array or Variable): The shifting parameter of input.
+        var (array or Variable): The square of scaling parameter of input.
         eps (float): Epsilon value for numerical stability.
 
     .. seealso::
        :func:`functions.batch_normalization`,
        :class:`links.BatchNormalization`
 
     """
-    return BatchNormalizationFunction(eps)(x, gamma, beta, mean, var)
+    return BatchNormalizationFunction(eps, mean, var, False,
+                                      0.0, use_cudnn)(x, gamma, beta)
```

Comment:
Yes, since the BN link does not need the gradients of `mean` and `var`, I think it should be possible to maintain backward compatibility of the BN function while also reducing computation by not computing these gradients. I will try this.